packaging/ubuntu-14.04/snapd.postrm

```diff
@@ -4,27 +4,24 @@ set -e
 
 systemctl_stop() {
     unit="$1"
-    if systemctl is-active -q "$unit"; then
-        echo "Stoping $unit"
+    for i in $(seq 10); do
+        if ! systemctl is-active -q "$unit"; then
+            echo "$unit is stopped."
+            break
+        fi
+        echo "Stoping $unit [attempt $i]"
         systemctl stop -q "$unit" || true
-    fi
+        sleep .1
+    done
 }
 
 if [ "$1" = "purge" ]; then
-    mounts=$(systemctl list-unit-files --full | grep '^snap[-.].*\.mount' | cut -f1 -d ' ')
-    services=$(systemctl list-unit-files --full | grep '^snap[-.].*\.service' | cut -f1 -d ' ')
-
-    for unit in $mounts; do
-        # ensure its really a snap mount unit or systemd unit
-        if ! grep -q 'What=/var/lib/snapd/snaps/' "/etc/systemd/system/$unit" && ! grep -q 'X-Snappy=yes' "/etc/systemd/system/$unit"; then
-            echo "Skipping non-snapd systemd unit $unit"
-            continue
-        fi
-        echo "Stopping $unit"
-        systemctl_stop "$unit"
-    done
+    # snap.mount.service is a trusty thing
+    systemctl_stop snap.mount.service
 
-    systemctl stop snap.mount.service || true
+    units=$(systemctl list-unit-files --full | grep '^snap[-.]' | cut -f1 -d ' ' | grep -vF snap.mount.service || true)
```

Comment:
Note that this doesn't catch units like `var-lib-snapd-snap-$SNAP_NAME`

Reply:
what uses those?

Reply:
Those get created by systemd if snap mount dir is different. I see that this is just for specific version of Ubuntu but I wanted to highlight this.